### PR TITLE
get with Int subscript

### PR DESCRIPTION
**Pull request by nhardman**
Add a convenience subscript method to allow access to elements with an integer index

### CODE REVIEW

File: Sources/TypeDecoder/OrderedDictionary.swift [hunk 1]
```diff
@@ -51,6 +51,17 @@ public struct OrderedDictionary<K: Hashable, V> {
         }
     }
 
+    /// subscript: get the value at the numeric index specified.
+    public subscript(index: Int) -> V? {
+        get {
```
Comment (ianpartridge) [inline Sources/TypeDecoder/OrderedDictionary.swift]: I don't think you need the `get{}` block.

File: Sources/TypeDecoder/OrderedDictionary.swift [hunk 2]
```diff
@@ -51,6 +51,17 @@ public struct OrderedDictionary<K: Hashable, V> {
...
+            if 0 ... self.keys.count - 1 ~= index {
```
Comment (ianpartridge) [inline Sources/TypeDecoder/OrderedDictionary.swift]: Can we use `keys.indices.contains(index)` instead?

File: Sources/TypeDecoder/OrderedDictionary.swift (hunk 1 above)
Comment (nhardman) [reply Sources/TypeDecoder/OrderedDictionary.swift]: done

File: Sources/TypeDecoder/OrderedDictionary.swift (hunk 2 above)
Comment (nhardman) [reply Sources/TypeDecoder/OrderedDictionary.swift]: done

**Pull request merged**